tests/e2e/ciliumnokp/ciliumnokp_test.go

```diff
@@ -0,0 +1,264 @@
+package ciliumnokp
+
+import (
+	"flag"
+	"fmt"
+
+	"os"
+	"strings"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/rancher/rke2/tests/e2e"
+)
+
+var nodeOS = flag.String("nodeOS", "generic/ubuntu2204", "VM operating system")
+var serverCount = flag.Int("serverCount", 1, "number of server nodes")
+var agentCount = flag.Int("agentCount", 1, "number of agent nodes")
+var ci = flag.Bool("ci", false, "running on CI")
+
+type objIP struct {
+	name string
+	ipv4 string
+	ipv6 string
+}
+
+// getClientIPs returns the IPs of the client pods
+func getClientIPs(kubeConfigFile string) ([]objIP, error) {
+	cmd := `kubectl get pods -l app=client -o=jsonpath='{range .items[*]}{.metadata.name}{" "}{.status.podIPs[*].ip}{"\n"}{end}' --kubeconfig=` + kubeConfigFile
+	return getObjIPs(cmd)
+}
+
+func getNodeIPs(kubeConfigFile string) ([]objIP, error) {
+	cmd := `kubectl get nodes -o jsonpath='{range .items[*]}{.metadata.name}{" "}{.status.addresses[?(@.type == "ExternalIP")].address}{"\n"}{end}' --kubeconfig=` + kubeConfigFile
+	return getObjIPs(cmd)
+}
+
+func getObjIPs(cmd string) ([]objIP, error) {
+	var objIPs []objIP
+	res, err := e2e.RunCommand(cmd)
+	if err != nil {
+		return nil, err
+	}
+	objs := strings.Split(res, "\n")
+	objs = objs[:len(objs)-1]
+
+	for _, obj := range objs {
+		fields := strings.Fields(obj)
+		if len(fields) > 2 {
+			objIPs = append(objIPs, objIP{name: fields[0], ipv4: fields[1], ipv6: fields[2]})
+		} else if len(fields) > 1 {
+			objIPs = append(objIPs, objIP{name: fields[0], ipv4: fields[1]})
+		} else {
+			objIPs = append(objIPs, objIP{name: fields[0]})
+		}
+	}
+	return objIPs, nil
+}
```

Comment:
Can you move these under `testutils.go`, we likely will use them again in other tests.

Reply:
good idea! I'll also change the dualStack test which is using this function as well, so that all use the ones in testutils. This PR will increase a bit the scope but I think it is worth it